charts/team-ns/templates/external-secrets.yaml

```diff
@@ -3,49 +3,80 @@
 {{- if get (index $v.apps "external-secrets") "enabled" }}
 {{- range $s := $v.secrets }}
 {{- $ns := index $s "namespace" | default $.Release.Namespace }}
-{{- $pathPrefix := printf "/secret/data/teams/team-%s" $v.teamId }}
+{{- $pathPrefix := printf "secret/data/teams/team-%s" $v.teamId }}
 {{- $vaultPath := printf "%s/%s" $pathPrefix $s.name | quote }}
-apiVersion: kubernetes-client.io/v1
+{{- $teamNamespacesCollection := print "team-" (include "helm-toolkit.utils.joinListWithSep" (dict "list" $v.teamIds "sep" ",team-")) }}
+
+apiVersion: external-secrets.io/v1beta1
+{{- if and (eq $v.teamId "admin") (or ($s.clusterWide | default false) ($s.teamWide | default false)) }}
+kind: ClusterExternalSecret
+{{- else }}
 kind: ExternalSecret
+{{- end }}
 metadata:
+  {{- if and (eq $v.teamId "admin") (or ($s.clusterWide | default false) ($s.teamWide | default false)) }}
+    {{- if $s.clusterWide | default false }}
```

Comment:
The use case for team-wide secrets seems to be very helpful as teams can have a shared pullSecret 🚀 

I am wondering if cluster-wide secret is needed, because there is already `globalPullSecret` implemented.
The `cluster-wide` secret  expands team-admin role to platform-admin role that can inject secrets to the platform namespaces. We have never considered team-admin to have any influence on platform namespaces. What do you think about it?

Reply:
I means more endpoints can be added. Some teams could use both dockerhub as well as github artifacts, and change platform images to use their own baked ones.

Reply:
But I do agree that the role should be otomi-admin for this, which should be added to api soon, as it is really easy to do. Until then I am fine shipping it like this. It introduces no security vulnerability, because if a team-admin was given a secret to do this, then he already has access and should be able to do so anyway. Now all other team-admins have access to that secret as well...so maybe turn it off until then?

I also will then allow a team admin to specify a list of teams instead of just checking "team wide".

Reply:
> I means more endpoints can be added. Some teams could use both dockerhub as well as github artifacts, and change platform images to use their own baked ones. 
I wouldn't be worried about that use case until we hear that someone really, really needs it - YAGNI and KISS

Reply:
Resolving it for now. We will observe if the cluster-wide secrets are  used and then make decision.